[ffigen] Can't find system headers with non-standard Xcode path

## Description

When using [Xcodes](https://github.com/XcodesOrg/XcodesApp) to manage Xcode installations, `dart run ffigen` fails with `stdio.h: file not found` error. Xcodes installs Xcode to paths like `/Applications/Xcode-26.1.1.app` instead of the standard `/Applications/Xcode.app`.

## Error Message

```
> dart run ffigen --config ffigen.yaml
Running in Directory: '/path/to/my_project'
Input Headers: [/path/to/my_project/src/example.h]
[SEVERE] : Header /path/to/my_project/src/example.h: Total errors/warnings: 1.
[SEVERE] :     /path/to/my_project/src/example.h:2:10: fatal error: 'stdio.h' file not found [Lexical or Preprocessor Issue]
[WARNING]: The compiler found warnings/errors in source files.
[WARNING]: This will likely generate invalid bindings.
[SEVERE] : Skipped generating bindings due to errors in source files. See https://github.com/dart-lang/native/blob/main/pkgs/ffigen/doc/errors.md.
```

## Environment

| Component  | Version                           |
| ---------- | --------------------------------- |
| Xcode      | 26.1.1 (installed via Xcodes app) |
| Xcode path | `/Applications/Xcode-26.1.1.app`  |
| ffigen     | 20.1.1                            |

## Steps to Reproduce

1. Install Xcode via Xcodes app (installs to `/Applications/Xcode-<version>.app`)
2. Create a project using ffigen with a C header that includes `<stdio.h>`
3. Run `dart run ffigen`

## Expected Behavior

ffigen should detect the correct SDK path regardless of where Xcode is installed.

## Actual Behavior

ffigen fails because it looks for headers in hardcoded paths under `/Applications/Xcode.app/`.

## Workaround

```bash
SDKROOT=$(xcrun --show-sdk-path) dart run ffigen
```

## Analysis

`lib/src/config_provider/path_finder.dart` has hardcoded paths:
- `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include`
- `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/`

These paths don't exist when Xcode is installed via Xcodes app.

Interestingly, `parser.dart` already uses `xcrun --show-sdk-path` for Objective-C mode (`_findObjectiveCSysroot()`), but this dynamic path resolution is not used for C mode.

--- a/pkgs/ffigen/CHANGELOG.md
+++ b/pkgs/ffigen/CHANGELOG.md
@@ -23,6 +23,8 @@
   always enabled.
 - Fix(https://github.com/dart-lang/native/issues/2877) 
   such that ObjCObject `isA` now accepts a nullable `ObjCObject?` and returns `false` when called with `null`, aligning its behavior with Dart’s `is`operator. 
+- Use `xcrun` for resolving macOS SDK paths, enabling support for non-standard
+  Xcode installations. ([#3134](https://github.com/dart-lang/native/issues/3134))
 
 ## 20.1.1
 

--- a/pkgs/ffigen/lib/src/config_provider/path_finder.dart
+++ b/pkgs/ffigen/lib/src/config_provider/path_finder.dart
@@ -10,13 +10,14 @@ import 'dart:io';
 import 'package:logging/logging.dart';
 import 'package:path/path.dart' as p;
 
+import 'utils.dart' show macSdkPath, xcodePath;
+
 /// This will return include path from either LLVM, XCode or CommandLineTools.
 List<String> getCStandardLibraryHeadersForMac(Logger logger) {
   final includePaths = <String>[];
 
   /// Add system headers.
-  const systemHeaders =
-      '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include';
+  final systemHeaders = p.join(macSdkPath, 'usr', 'include');
   if (Directory(systemHeaders).existsSync()) {
     logger.fine('Added $systemHeaders to compiler-opts.');
     includePaths.add('-I$systemHeaders');
@@ -24,9 +25,15 @@ List<String> getCStandardLibraryHeadersForMac(Logger logger) {
 
   /// Find headers from XCode or LLVM installed via brew.
   const brewLlvmPath = '/usr/local/opt/llvm/lib/clang';
-  const xcodeClangPath =
-      '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/';
-  const searchPaths = [brewLlvmPath, xcodeClangPath];
+  final xcodeClangPath = p.join(
+    xcodePath,
+    'Toolchains',
+    'XcodeDefault.xctoolchain',
+    'usr',
+    'lib',
+    'clang',
+  );
+  final searchPaths = [brewLlvmPath, xcodeClangPath];
   for (final searchPath in searchPaths) {
     if (!Directory(searchPath).existsSync()) continue;
 

